client/my-sites/email/titan-complete-mailbox-setup/index.js

```diff
@@ -0,0 +1,253 @@
+/**
+ * External dependencies
+ */
+import { Button, Card } from '@automattic/components';
+import page from 'page';
+import React, { useCallback, useState } from 'react';
+import PropTypes from 'prop-types';
+import { useDispatch, useSelector } from 'react-redux';
+import { useTranslate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import AddEmailAddressesCardPlaceholder from 'calypso/my-sites/email/gsuite-add-users/add-users-placeholder';
+import {
+	areAllMailboxesValid,
+	buildNewTitanMailbox,
+	decorateMailboxWithAvailabilityError,
+	validateMailboxes,
+} from 'calypso/lib/titan/new-mailbox';
+import DocumentHead from 'calypso/components/data/document-head';
+import EmailHeader from 'calypso/my-sites/email/email-header';
+import {
+	emailManagement,
+	emailManagementCompleteTitanMailboxSetup,
+} from 'calypso/my-sites/email/paths';
+import { errorNotice } from 'calypso/state/notices/actions';
+import getCurrentRoute from 'calypso/state/selectors/get-current-route';
+import { getDomainsBySiteId, hasLoadedSiteDomains } from 'calypso/state/sites/domains/selectors';
+import { getSelectedDomain } from 'calypso/lib/domains';
+import { getSelectedSite } from 'calypso/state/ui/selectors';
+import { hasTitanMailWithUs } from 'calypso/lib/titan';
+import HeaderCake from 'calypso/components/header-cake';
+import Main from 'calypso/components/main';
+import PageViewTracker from 'calypso/lib/analytics/page-view-tracker';
+import QuerySiteDomains from 'calypso/components/data/query-site-domains';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import TitanNewMailboxList from 'calypso/my-sites/email/titan-add-mailboxes/titan-new-mailbox-list';
+import { useCreateTitanMailboxMutation } from 'calypso/data/emails/use-create-titan-mailbox-mutation';
+import { useGetTitanMailboxAvailability } from 'calypso/data/emails/use-get-titan-mailbox-availability';
+
+const getMailboxUserName = ( mailbox ) => mailbox?.mailbox?.value;
```

Comment:
I believe this should be named `getMailboxName` to be consistent with the other parts of the code you're introducing (it doesn't return `mailbox.name.value` anyway).

Reply:
I prefer this specificity as there's also a mailbox property called `name`

Reply:
Oh, I see this where the confusion comes from now:

https://github.com/Automattic/wp-calypso/blob/4940d44a93cc23a6b293ccac4ddb1c63533924b6/client/lib/titan/new-mailbox.js#L360-L361